src/Dropdown/MultiDropdownWithSelectAll.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from "react";
+import Select, { components } from "react-select";
+
+import Creatable from "react-select/lib/Creatable";
+import { Props } from "react-select/lib/Creatable";
+import * as cx from "classnames";
+import {
+  CommonProps,
+  defaultProps,
+  defaultComponents,
+  defaultStyle,
+  getCommonClassnames,
+  DefaultProps
+} from "./commons";
+import { GroupType, ValueType } from "react-select/lib/types";
+
+export function allSelected<OptionType = never>(): SelectAllValue<OptionType> {
+  return { type: "AllSelected" };
+}
+
+export function someSelected<OptionType>(values: OptionType[]) {
```

Comment:
just for coherence, i would add the return type annotation: `SelectAllValue<OptionType>`